File: tests/test_saphyr.py
"""
Test saphyr-parser vs yaml-rust2
"""

import saphyr_parser
import yaml_rust2


def test_basic_parse():
    """Test basic parsing"""
    yaml_str = "name: John\nage: 30"

    # saphyr-parser
    print("=== saphyr-parser ===")
    try:
        docs = saphyr_parser.load(yaml_str)
        print(f"Success: {docs}")
    except Exception as e:
        print(f"Error: {e}")

    # yaml-rust2
    print("\n=== yaml-rust2 ===")
    try:
        docs = yaml_rust2.load(yaml_str)
        print(f"Success: {docs}")
    except Exception as e:
        print(f"Error: {e}")


if __name__ == '__main__':
    test_basic_parse()

File: src/test_saphyr.rs
/// Test saphyr-parser integration
#[cfg(test)]
mod saphyr_tests {
    use saphyr_parser::{Event, EventReceiver, Parser};

    struct EventSink<'a> {
        events: Vec<Event<'a>>,
    }

    impl<'a> EventReceiver<'a> for EventSink<'a> {
        fn on_event(&mut self, ev: Event<'a>) {
            self.events.push(ev);
        }
    }

    #[test]
    fn test_basic_parse() {
        let yaml = "name: John\nage: 30";
        let mut sink = EventSink { events: Vec::new() };
        let mut parser = Parser::new_from_str(yaml);
        parser.load(&mut sink, true).unwrap();

        println!("Events:");
        for event in &sink.events {
            println!("  {:?}", event);
        }

        assert!(!sink.events.is_empty());
    }

    #[test]
    fn test_complex_yaml() {
        let yaml = r#"
# Comment
defaults: &defaults
  timeout: 30

production:
  <<: *defaults
  host: prod.com
"#;
        let mut sink = EventSink { events: Vec::new() };
        let mut parser = Parser::new_from_str(yaml);
        parser.load(&mut sink, true).unwrap();

        println!("Complex YAML events:");
        for event in &sink.events {
            println!("  {:?}", event);
        }

        assert!(!sink.events.is_empty());
    }
}
